File: Solutions/0876- Middle of the Linked List/Solution.java
class Solution {
    public ListNode middleNode(ListNode head) {
        var slow = head;
        var fast = head;

        while (fast.next != null && fast.next.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }

        return fast.next != null ? slow.next : slow;
    }
}

File: Solutions/0876- Middle of the Linked List/Solution.py
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        slow = fast = head

        while fast.next is not None and fast.next.next is not None:
            slow = slow.next
            fast = fast.next.next

        return slow.next if fast.next is not None else slow